### PR TITLE
Zookeeper upgrade

### PR DESCRIPTION
**Pull request by farajn9**
Added ZooKeeper 3.5.8 version templates and Dockerfiles
Adapt Functional tests

### CODE REVIEW

File: docker/zookeeper/versions.yaml [hunk 1]
```diff
@@ -16,6 +16,25 @@ _variables:
...
+    from: marketplace.gcr.io/google/debian10
+    packages:
+      zookeeper:
+        version: '3.5.8'
+        gpg: '52A7EA3EECAE05B0A8306471790761798F6E35FC'
+    excludeTests:
+    - tests/functional_tests/exporter_test.yaml
 
   # ZooKeeper image
   - dir: '3/debian9/3.4'
```
Comment (ovk6) [inline docker/zookeeper/versions.yaml:54]: Please move `from: gcr.io/google-appengine/openjdk` here and remove variables sections, as it is not used anywhere else

File: docker/zookeeper/tests/functional_tests/exporter_test.yaml [hunk 2]
```diff
@@ -32,5 +32,5 @@ tests:
   - http://localhost:9141/metrics
   expect:
     stdout:
-      matches: '(?m)^zk_up 1$'
+      matches: 'zk_up'
```
Comment (ovk6) [inline docker/zookeeper/tests/functional_tests/exporter_test.yaml]: Just `zk_up` can also match `zk_up 0` which indicates failure.
Should be `zk_up 1`

File: docker/zookeeper/templates/debian10/Dockerfile.template [hunk 3]
```diff
@@ -0,0 +1,95 @@
+FROM {{ .From }}
+
+{{- $zookeeper := index .Packages "zookeeper" }}
```
Comment (ovk6) [inline docker/zookeeper/templates/debian10/Dockerfile.template]: As we discussed previously all variables should be defined at the beginning of file, eg. https://github.com/GoogleCloudPlatform/click-to-deploy/blob/master/docker/drupal/templates/Dockerfile.template#L1

File: docker/zookeeper/README.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ Learn more about ZooKeeper in [official documentation](https://zookeeper.apache.
 
 ## Upstream
 
+- Source for [ZooKeeper 3.5+](https://github.com/31z4/zookeeper-docker)
+
 - Source for [ZooKeeper](https://github.com/kubernetes-retired/contrib/blob/master/statefulsets/zookeeper/Dockerfile)
```
Comment (ovk6) [inline docker/zookeeper/README.md]: Source for [ZooKeeper 3.4 ...]

File: docker/zookeeper/versions.yaml [hunk 5]
```diff
@@ -16,6 +16,25 @@ _variables:
   from: &from gcr.io/google-appengine/openjdk
 
 versions:
+  # ZooKeeper image
+  - dir: '3/debian10/3.5'
```
Comment (ovk6) [inline docker/zookeeper/versions.yaml:37]: Why not updating directly to 3.6?

File: docker/zookeeper/3/debian10/3.5/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    ZOO_AUTOPURGE_SNAPRETAINCOUNT=3 \
+    ZOO_MAX_CLIENT_CNXNS=60 \
+    ZOO_STANDALONE_ENABLED=true \
+    ZOO_ADMINSERVER_ENABLED=true
+
+# Add a user with an explicit UID/GID and create necessary directories.
+RUN set -eux; \
+    groupadd -r zookeeper --gid=1000; \
+    useradd -r -g zookeeper --uid=1000 zookeeper; \
+    mkdir -p "${ZOO_DATA_LOG_DIR}" "${ZOO_DATA_DIR}" "${ZOO_CONF_DIR}" "${ZOO_LOG_DIR}"; \
```
Comment (ovk6) [inline docker/zookeeper/3/debian10/3.5/Dockerfile:22]: IMHO better to leave original style in Dockerfile and entrypoint if it comes to variables ${} etc and add only necessary changes. Will be easier to track any updates in upstream.

**Review by ovk6 — CHANGES_REQUESTED** [review]

File: docker/zookeeper/versions.yaml (hunk 5 above)
Comment (farajn9) [reply docker/zookeeper/versions.yaml:37]: This version is needed for Solr, and for 3.6 I will add it later.

File: docker/zookeeper/3/debian10/3.5/Dockerfile (hunk 6 above)
Comment (farajn9) [reply docker/zookeeper/3/debian10/3.5/Dockerfile:22]: Thanks for the comment but I think with this standard it looks better

**Review by armandomiani — APPROVED** [review]

File: docker/zookeeper/3/debian10/3.5/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,95 @@
+
```
Comment (aav66) [inline docker/zookeeper/3/debian10/3.5/Dockerfile]: remove empty line

File: docker/zookeeper/3/debian10/3.5/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
```
Comment (aav66) [inline docker/zookeeper/3/debian10/3.5/Dockerfile]: remove empty line

File: docker/zookeeper/3/debian10/3.6/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,95 @@
+
```
Comment (aav66) [inline docker/zookeeper/3/debian10/3.6/Dockerfile]: remove empty line

File: docker/zookeeper/3/debian10/3.6/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
```
Comment (aav66) [inline docker/zookeeper/3/debian10/3.6/Dockerfile]: remove empty line

**Review by aav66 — CHANGES_REQUESTED** [review]

**Pull request merged**